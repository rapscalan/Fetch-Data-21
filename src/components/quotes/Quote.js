import React from 'react';
import PropTypes from 'prop-types';

const Quote = ({ text, name, img }) => (
  <figure>
    <img src={img} />
    <figcaption>&quot;{text}&quot;{name ? ` -- by ${name}` : ''}</figcaption>
  </figure>
);

Quote.propTypes = {
  text: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  name: PropTypes.string
};

export default Quote;
