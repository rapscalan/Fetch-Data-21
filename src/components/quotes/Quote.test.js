import React from 'react';
import { shallow } from 'enzyme';
import Quote from './Quote';

describe('Quote component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<Quote text="My Quote" name="Me" img="/path/to/image" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('matches a snapshot when name isn\'t provided', () => {
    const wrapper = shallow(<Quote text="My Quote" img="/path/to/image" />);
    expect(wrapper).toMatchSnapshot();
  });
});
