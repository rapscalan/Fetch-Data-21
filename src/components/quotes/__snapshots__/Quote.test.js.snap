// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component matches a snapshot 1`] = `
<figure>
  <img
    src="/path/to/image"
  />
  <figcaption>
    "
    My Quote
    "
     -- by Me
  </figcaption>
</figure>
`;

exports[`Quote component matches a snapshot when name isn't provided 1`] = `
<figure>
  <img
    src="/path/to/image"
  />
  <figcaption>
    "
    My Quote
    "
  </figcaption>
</figure>
`;
